src/depth_packet_processor.cpp

```diff
@@ -98,4 +104,51 @@ void DepthPacketProcessor::setFrameListener(libfreenect2::FrameListener *listene
   listener_ = listener;
 }
 
+  DumpDepthPacketProcessor::DumpDepthPacketProcessor() : depth_frame(new Frame(512, 424, 4)), ir_frame(new Frame(512, 424,4)) {
+}
+
+DumpDepthPacketProcessor::~DumpDepthPacketProcessor(){
+  delete depth_frame;
+  delete ir_frame;
+}
+
+void DumpDepthPacketProcessor::process(const DepthPacket &packet) {
+    std::stringstream name;
+    name << packet.timestamp << "_" << packet.sequence << "_" << packet.buffer_length << ".depth";
+  
+    std::ofstream file(name.str().c_str());
+    file.write(reinterpret_cast<char *>(packet.buffer), packet.buffer_length);
+    file.close();
+
+    std::stringstream json;
+    json << "{" << std::endl;
+    json << "\"timestamp\": " << packet.timestamp << "," << std::endl;
+    json << "\"sequence\": " << packet.sequence << "," << std::endl;
+
+    timespec now;
+    clock_gettime(CLOCK_REALTIME, &now);
```

Comment:
thanks for the thorough explanation.  I'll use std::chrono and deal with it being missing.